pkg/validate/astrolabevalidator.go

```diff
@@ -0,0 +1,76 @@
+package validate
+
+import (
+	"fmt"
+	"log"
+)
+
+type AstrolabeCompatibilityValidator struct{}
+
+func (v *AstrolabeCompatibilityValidator) IsPluginCompatibleWithPlatform(platformVersion string, pluginId string, pluginVersion string, repos []PluginRepository) (bool, error) {
+	var metadata *CompatibilityMetadata
+	for _, e := range repos {
+		m, err := e.getCompatibilityMetadata(pluginId)
+		if err != nil {
+			log.Printf("error when geting compatibility metadata: %s", err)
+		} else {
+			metadata = m
+			break
+		}
+	}
+
+	if metadata == nil {
+		return false, fmt.Errorf("compatibility metadata not found in repositories")
+	}
+
+	tests, err := metadata.getCompatibilityTests(pluginVersion)
+	if err != nil {
+		return false, err
+	}
+
+	for _, e := range tests {
+		if e.containsPlatformVersion(platformVersion) {
+			if e.Status == "success" {
+				return true, nil
+			} else {
+				return false, nil
+			}
+		}
+	}
+
+	return false, fmt.Errorf("could not find compatibility tests with platform: %s for plugin %s@%s", platformVersion, pluginId, pluginVersion)
```

Comment:
What do you think about having three different outcomes here (instead of a boolean)? Compatible, Not Compatible, and Unknown.

Reply:
I think would be a good improvement, currently when the function returns an error it can be interpreted as Unknown but having the three states that you mention will help the consumers to have a clear way on what was the verdict